runtime/src/iree/hal/drivers/hip/hip_device.h

```diff
@@ -42,6 +42,9 @@ iree_status_t iree_hal_hip_device_create_stream_command_buffer(
 // contexts and the context may be in use on other threads.
 hipCtx_t iree_hal_hip_device_context(iree_hal_device_t* device);
 
+// Returns the hip device for the given iree_hal_device.
+hipDevice_t iree_hal_hip_device_get_device(iree_hal_device_t* device);
```

Comment:
instead of exposing this you can pass it as an argument to the pending actions queue - we want to reserve the ability to have multiple hipDevice_t's per single logical IREE HAL device in the future and this prevents that (we'd want one pending action queue per device)

Reply:
Sure, I had actually assumed the opposite, where we would have a single pending_actions_queue and multiple devices, that actually things a bit for binding the device to a thread.